src/udev/ata_id/ata_id.c

```diff
@@ -331,7 +332,7 @@ static int disk_identify(int fd,
                 return r;
 
         /* SPC-4, section 6.4.2: Standard INQUIRY data */
-        peripheral_device_type = inquiry_buf[0] & 0x1f;
+        peripheral_device_type = *out_peripheral_device_type = inquiry_buf[0] & 0x1f;
```

Comment:
Please do not update the argument to store a result in intermediate stage of the function.
Instead, please update it on success.
```c
        if (all_nul_bytes)
                return log_debug_errno(SYNTHETIC_ERRNO(EIO), "IDENTIFY data is all zeroes.");

        if (ret_peripheral_device_type)
                *ret_peripheral_device_type = peripheral_device_type;

        return 0;
}

```